src/Components/Analyzers/src/ComponentFacts.cs

```diff
@@ -85,5 +102,76 @@ public static bool IsCascadingParameter(ComponentSymbols symbols, IPropertySymbo
 
             return property.GetAttributes().Any(a => a.AttributeClass == symbols.CascadingParameterAttribute);
         }
+
+        public static bool IsComponent(ComponentSymbols symbols, INamedTypeSymbol type)
+        {
+            if (symbols is null)
+            {
+                throw new ArgumentNullException(nameof(symbols));
+            }
+
+            if (type is null)
+            {
+                throw new ArgumentNullException(nameof(type));
+            }
+
+            if (!IsAssignableFrom(symbols.IComponentType, type))
+            {
+                return false;
+            }
+
+            return true;
+        }
+
+        public static bool IsAssignableFrom(ITypeSymbol source, ITypeSymbol target)
+        {
+            if (source is null)
+            {
+                throw new ArgumentNullException(nameof(source));
+            }
+
+            if (target is null)
+            {
+                throw new ArgumentNullException(nameof(target));
+            }
+
+            if (source == target)
+            {
+                return true;
+            }
+
+            if (source.TypeKind == TypeKind.Interface)
+            {
+                foreach (var @interface in target.AllInterfaces)
+                {
+                    if (source == @interface)
+                    {
+                        return true;
+                    }
+                }
+
+                return false;
+            }
+
+            foreach (var type in target.GetTypeHierarchy())
+            {
+                if (source == type)
+                {
+                    return true;
+                }
+            }
+
+            return false;
```

Comment:
We should see if this can be done with `ClassifyConversion` instead of reimplementing this logic.